framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -3442,4 +3547,32 @@ public ProcessingEnvironment getProcessingEnv() {
     public CFContext getContext() {
         return checker;
     }
+
+    /**
+     * Factory method to reflectively set {@link AnnotatedTypeFactory#viewpointAdaptor}. By default,
+     * this method gets the name of type system from the canonical name of "this", and concatenate
+     * it with "ViewpointAdaptor". If the concrete type system has a different naming mechanism,
+     * then developer of that type system should override this method to instantiate it correctly.
+     * If the target class can be instantiated successfully, it's used. Otherwise, internal error is
+     * triggered and will be error aborted.
+     *
```

Comment:
The parameter needs to be documented. Why pass the value of `checker.withViewpointAdaptation()` as a parameter at all? Can't each implementation choose whether they care about that value?